src/material/datepicker/calendar-body.ts

```diff
@@ -451,12 +451,22 @@ export class MatCalendarBody<D = any> implements OnChanges, OnDestroy, AfterView
     }
 
     if (this.startValue === value && this.endValue === value) {
-      return `${this._startDateLabelId} ${this._endDateLabelId}`;
+      return `${this._startDateLabelId}-${this._endDateLabelId}`;
```

Comment:
Spaces are used to delimit the IDs in the described by. Otherwise we're seeing invalid IDs like `mat-calendar-body-3-start-date-mat-calendar-body-3-end-date`